expression/scalar_function.go

```diff
@@ -76,9 +77,15 @@ func NewFunction(ctx context.Context, funcName string, retType *types.FieldType,
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	var tp *types.FieldType
+	if retType == nil {
```

Comment:
I'm not really sure, it seems not that much when tracing the caller of NewFunction.